colour/colorimetry/spectrum.py

```diff
@@ -843,11 +843,19 @@ def shape(self) -> SpectralShape:
                 f"minimum interval!"
             )
 
-        return SpectralShape(
-            min(self.wavelengths),
-            max(self.wavelengths),
-            min(wavelengths_interval),
-        )
+        # If wavelegths is sorted, then go fast. Otherwise compute the shape.
+        if np.all(self.wavelengths[:-1] <= self.wavelengths[1:]):
```

Comment:
At the point where we try to get some speed, should not we also store `self.wavelengths` into an intermediate variable so that we do not access and copy it two times (see `colour.Signal.range` property)?

What about this: `np.all(np.diff(wavelengths) >= 0)` ?

Reply:
I tested diff and it's actually slower than this. I will check memcopys though.

Reply:
Double checked. I couldn't find a way to eliminate any further copies.